### ISSUE TITLE
When UseWPF set True in a WinForms. Proj fails to recognize ApplicationConfiguration bootstrap code

### ISSUE THREAD

**Issue by Njoro007** [dotnet/winforms#6947]
### .NET version

6.0.3

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Worked without any issues in NET 5 and Net Framework

### Issue description

In a WinForms desktop application targeting .NET 6, the [UseWindowsForms] element in the .proj file is set to True, and a bootstrap ApplicationConfiguration.Initialize(); is generated and called from the startup file Program.

Now I want to enable WPF by setting [EnableWPF] to True. When I do that and build I get the following error in the output log

` Program.cs(13,13,13,37): error CS0103: The name 'ApplicationConfiguration' does not exist in the current context
1>Done building project "WinFormsApp1_s2dcxfug_wpftmp.csproj" -- FAILED. `

**My workaround for a net6 WinForm Project with WPF Enabled is editing the Program.CS to code below**
`
//ApplicationConfiguration.Initialize();
Application.SetHighDpiMode(HighDpiMode.SystemAware);
Application.EnableVisualStyles();
Application.SetCompatibleTextRenderingDefault(false);
Application.Run(new Form1());
`

### Steps to reproduce

**To Reproduce**

- Create a new Windows Forms project in VS 2022 targeting .NET 6.
- Go to proj properties and check the Enable WPF for this project check box
- Add a WPF window to the project
- Build

**Comment by dreddy-work** [comment]
@RussKie, can you take a look at this?

**Comment by RussKie** [comment]
@Njoro007 please provide a simple repro that manifest the issue.

**Comment by Njoro007** [comment]
To Reproduce

- Create a new Windows Forms project in VS 2022 targeting .NET 6.

- Go to Application/Proj Properties, under Application **check** the _Enable WPF for this project_ check box

- Add a WPF window to the project

- and then Build. (Error will show)

**Comment by RussKie** [comment]
I am unable to repro it. 
![image](https://user-images.githubusercontent.com/4403806/162332638-21279a50-8dab-4bbe-9d5c-b643577b3735.png)

**Comment by Njoro007** [comment]
Added a WPF Window/Page?...I have done a recording

https://user-images.githubusercontent.com/22431006/162335073-20ec867f-dd95-447d-8e2f-d145384f6d9b.mp4

**Comment by RussKie** [comment]
Thank you for the video. I can repro it now.

**Comment by RussKie** [comment]
I've looked through the [binlog](https://github.com/dotnet/winforms/files/8456422/msbuild.zip), and there's a second compilation pass for a temp WPF project:
![Untitled](https://user-images.githubusercontent.com/4403806/162560731-eea88b31-eef6-4bf1-a224-83bf252edd42.png)

I _guess_ the Windows Forms source generator isn't run for the temp project, and the temp project can't resolve `ApplicationConfiguration.Initialize()` call, and this is the cause of the failure.

The Windows Forms SG generates the body of `ApplicationConfiguration.Initialize()`, but we don't emit it in a form a file, and hence we can't pass it on to the WPF temp project for a compilation.
https://github.com/dotnet/winforms/blob/c402e2a21cc3d6ab1c3fe6209ff829361aa596b9/src/System.Windows.Forms.Analyzers.CSharp/src/System/Windows/Forms/Generators/ApplicationConfigurationGenerator.cs#L50

@sharwell @chsienki @jmarolf I think I need you help here. 
* How can I check/determine whether the Windows Forms SG is run by the temp project?
* Can we save `ApplicationConfiguration.g.cs` as fie?

Thanks

**Comment by chsienki** [comment]
Hmm, we've seen this problem in a few different iterations. 

https://github.com/dotnet/roslyn/issues/60257
https://github.com/dotnet/wpf/issues/4421
https://github.com/dotnet/roslyn/issues/46300

Basically all of these occur because the 'inner build' isn't including the required source generators. I think it was generally fixed for WPF, but my best guess is that the winforms generator is somehow being conditionally included on the `<UseWinForms>` project property and the temp WPF project doesn't have that set.

**Comment by chsienki** [comment]
@RussKie Yeah, the `ResolveTargetingPackAssets` target isn't getting run by the WPF inner build, so the winforms generator never gets picked up by that build. I'd suggest starting with WPF to see why that isn't run / if it can be.

**Comment by RussKie** [comment]
Our source generator is only included for Windows Forms apps:
https://github.com/dotnet/winforms/blob/8e757eb35354dd9f0d6297ff721dd2c1f773b59d/pkg/Microsoft.Private.Winforms/sdk/dotnet-windowsdesktop/System.Windows.Forms.FileClassification.props#L13-L14
dotnet\packs\Microsoft.WindowsDesktop.App.Ref\\<version>\data\FrameworkList.xml is then generated from this list, which is then used by `ResolveTargetingPackAssets` target to resolve what analyzers need to be included for a project.
I don't believe it'd be right to include Windows Forms specific source generators and analyzer for a WPF application.

My thinking is though along the lines of that the source generated content should be present on disk, so it can be included in the compilation process. In a way similar to other generated (though not source generated) files, e.g., `obj\Debug\net6.0-windows\<project name>.AssemblyInfo.cs` or `obj\Debug\net6.0-windows\<project name>.GlobalUsings.g.cs`.

**Comment by RussKie** [comment]
Looking through the binlog stumbled upon this:
![image](https://user-images.githubusercontent.com/4403806/166612380-bc47ad63-d061-4c14-b452-6787c91fd5a7.png)

Digging in further...
